experimental/tzif/src/parse/posix.rs

```diff
@@ -0,0 +1,768 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::{
+    core::{
+        ext::{
+            byte::ParseBytes,
+            chain::Chain,
+            default::OrDefaultTo,
+            inspect::Inspect,
+            many::ParseMany,
+            map::{MapInner, MapValue, TryMapValue},
+            maybe::MaybeParse,
+            require::Require,
+            then::{OnSuccess, Then},
+        },
+        ParseInput, ParseResult, Parsed,
+    },
+    data::{
+        posix::{DstTransitionInfo, PosixTzString, TransitionDate, TransitionDay, ZoneVariantInfo},
+        time::{Hours, Minutes, Seconds},
+    },
+};
+use eyre::Context;
+
+impl<T, Source> PosixTzStringParser<Source> for T
+where
+    Source: Clone + ParseInput<u8, Vec<u8>, Source>,
+    T: ParseInput<u8, Vec<u8>, Source>,
+{
+}
```

Comment:
Suggestion: An empty blanket impl like this suggests to me that you should instead do a wrapper struct.

```rust
pub struct PosixTzStringParser<Source: Clone + ParseInput<u8, Vec<u8>, Source>> {
    source: Source
}
```

And them impl all your methods on the struct.

I think there are several places where you should be able to do this.

Reply:
Yeah this is what I meant when I said I wasn't clear on why some of these things needed to be traits.